src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test.Common/LanguageServerTestBase.cs

```diff
@@ -17,20 +18,31 @@ public abstract class LanguageServerTestBase
     {
         public LanguageServerTestBase()
         {
-            Dispatcher = new TestProjectSnapshotManagerDispatcher();
+#pragma warning disable CS0618 // Type or member is obsolete
+            LegacyDispatcher = new TestProjectSnapshotManagerDispatcher();
+#pragma warning restore CS0618 // Type or member is obsolete
             FilePathNormalizer = new FilePathNormalizer();
             var logger = new Mock<ILogger>(MockBehavior.Strict).Object;
             Mock.Get(logger).Setup(l => l.Log(It.IsAny<LogLevel>(), It.IsAny<EventId>(), It.IsAny<It.IsAnyType>(), It.IsAny<Exception>(), It.IsAny<Func<It.IsAnyType, Exception, string>>())).Verifiable();
             Mock.Get(logger).Setup(l => l.IsEnabled(It.IsAny<LogLevel>())).Returns(false);
             LoggerFactory = Mock.Of<ILoggerFactory>(factory => factory.CreateLogger(It.IsAny<string>()) == logger, MockBehavior.Strict);
+            Dispatcher = new LSPProjectSnapshotManagerDispatcher(LoggerFactory);
         }
 
+        // This is marked as legacy because in its current form it's being assigned a "TestProjectSnapshotManagerDispatcher" which takes the
+        // synchronization context from the constructing thread and binds to that. We've seen in XUnit how this can unexpectedly lead to flaky
+        // tests since it doesn't actually replicate what happens in real scenariso (a separate dedicated dispatcher thread). If you're reading
```

Comment:
Helpful comment! Small nit:
```suggestion
        // tests since it doesn't actually replicate what happens in real scenario (a separate dedicated dispatcher thread). If you're reading
```